content/en/blog/2023/jmx-metric-insight/index.md

```diff
@@ -0,0 +1,229 @@
+create a topic and use the Kafka console producer and console consumer. Create
+Kafka Topic:
+
+```shell
+kafka-topics --create --bootstrap-server localhost:9092 --replication-factor 1 --partitions 1 --topic my-test-topic
+```
+
+Start Kafka console producer to produce messages to the topic we created:
+
+```shell
+kafka-console-producer --broker-list localhost:9092 --topic test
+>First message
+>Second message
+```
+
+Now we will start the Kafka console consumer which will consume messages from
+the topic from the beginning:
+
+```shell
+kafka-console-consumer --bootstrap-server localhost:9092 --topic test --from-beginning
+First message
+Second message
+```
+
+If we are able to see the two messages received by the consumer, that verifies
+that our Kafka installation is working as expected.
+
+### Export metrics to Prometheus
+
+The metrics can be exported by any of the supported metric exporters, to a
+backend of your choice. The full list of exporters and their configuration
+options can be found
+[here](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#exporters).
+For instance, you can export the metrics to an OTel collector using the OTLP
+exporter, perform some processing and then consume the metrics on a backend of
+your choice. In this example for the sake of simplicity, we are directly
+exporting the metrics to Prometheus.
+
+We will visualise the metrics on a Grafana dashboard using Prometheus as the
+data source. In this demo, we’ll deploy Prometheus on Docker. We can create a
+`prometheus.yml` file containing this minimal configuration:
+
+```yaml
+global:
+  scrape_interval: 10s
+  evaluation_interval: 10s
+
+scrape_configs:
+  - job_name: "my-kafka-broker"
+    scrape_interval: 5s
+    static_configs:
+      - targets: ["host.docker.internal:9464"]
+```
+
+Then run the command below to deploy Prometheus on Docker:
+
+```shell
+docker run -d \
+    -p 9090:9090 \
+    -v path/to/prometheus.yml:/etc/prometheus/prometheus.yml \
+    prom/prometheus
+```
+
+The Prometheus container should be running now. You can now navigate to
+[http://localhost:9090](http://localhost:9090) and explore the Prometheus
+dashboard. Here we are viewing the metric `kafka_request_count_total` on
+Prometheus.
+
+![A sample Kafka Broker metric shown on Prometheus](prometheus.png)
+
+More installation options for Prometheus can be found
+[here](https://prometheus.io/docs/prometheus/latest/installation/).
+
+### View the metrics on a Grafana Dashboard
+
+Now, we are going to visualise the Prometheus metrics in a Grafana dashboard. To
+do that, first, pull the Grafana docker image using the following command:
+
+```shell
+docker run -d -p 3000:3000 grafana/grafana
+```
+
+You can now navigate to [http://localhost:3000](http://localhost:3000) and
+explore the Grafana home page. Click on Add Data Source and select Prometheus.
+Add the HTTP URL, default is `http://localhost:9090`. After that we can create
```

Comment:
Can this be a link too (it might slightly easier to copy that way)?

```suggestion
Add the HTTP URL, default is <http://localhost:9090>. After that we can create
```